client/actions/shield-study/index.js

```diff
@@ -1,13 +1,34 @@
 import { Action, registerAction } from '../utils';
 
 export default class ShieldStudyAction extends Action {
-  constructor(normandy, recipe) {
-    super(normandy, recipe);
-    this.storage = normandy.createStorage(recipe.id);
+  setHasBeenShown() {
+    this.storage.setItem('studyHasBeenShown', true);
+  }
+
+  async getHasBeenShown() {
+    const hasBeenShown = await this.storage.getItem('studyHasBeenShown');
+    return hasBeenShown;
   }
 
   async execute() {
+    try {
+      this.storage = await this.normandy.createStorage(this.recipe.id);
+    } catch (error) {
+      throw new Error(error);
```

Comment:
You don't need to catch and then rethrow the error. You can just let it fail, leaving out all the try/catch stuff.